### PR TITLE
New contracts setup 

### PR DESCRIPTION
**Pull request by Bromel777**

### CODE REVIEW

**Review by GusevTimofey — APPROVED** [review]

File: amm-executor/src/Spectrum/Executor/EventSink/Handlers/Orders.hs [hunk 1]
```diff
@@ -40,41 +40,71 @@ import Spectrum.Executor.EventSource.Data.TxContext
...
+       slotsTime = secondsToNominalDiffTime . fromIntegral $ unSlotNo slotNo
+       txTime    = addUTCTime slotsTime (getSystemStart systemStart)
+    if (diffUTCTime currentTime txTime > orderLifetime)
+      then infoM ("Tx is outdated : " ++ show txId) >> pure Nothing
+      else (parseOrder logging `traverse` txOutputs) >>= foldM (process txTime) Nothing           
       where
-        process _ ordM = do
-          ts <- getCurrentTime
-          mapM publish $ ordM <&> flip PendingOrder ts
+        process oTime _ ordM = (\order -> liftIO (signalQSem syncSem) >> publish order) `mapM` (ordM <&> flip PendingOrder oTime)
```
Comment (oskin1) [inline amm-executor/src/Spectrum/Executor/EventSink/Handlers/Orders.hs:67]: Let's not use `traverse` and `mapM` in infix notation as they are not supposed to be used like that.

File: amm-executor/src/Spectrum/Executor/EventSink/Handlers/Pools.hs [hunk 2]
```diff
@@ -21,13 +22,40 @@ import Spectrum.Executor.Data.State
...
   _ -> pure Nothing
+
+parsePool :: (MonadIO m) => Logging m -> FullTxOut -> m (Maybe (Confirmed (OnChain Pool)))
+parsePool Logging{..} out@FullTxOut{..} = do
+  pValidator <- poolValidator
+  let
+    pool        = parseFromLedger out
+    poolAddress = mkValidatorAddress pValidator
+  if (fullTxOutAddress == poolAddress) then
+    case pool of
```
Comment (oskin1) [inline amm-executor/src/Spectrum/Executor/EventSink/Handlers/Pools.hs:54]: Please keep `if..then..else..` formatting consistent

File: amm-executor/src/Spectrum/Executor/EventSink/Handlers/Pools.hs [hunk 3]
```diff
@@ -21,13 +22,40 @@ import Spectrum.Executor.Data.State
...
+      Just a    -> do
+        infoM ("Pool found in: " ++ show out)
+        pure $ Just $ Confirmed a
+      _                                 -> do
```
Comment (oskin1) [inline amm-executor/src/Spectrum/Executor/EventSink/Handlers/Pools.hs:58]: What's wrong with `                      -> do`?

File: amm-executor/src/Spectrum/Executor/OrdersExecutor/Process.hs [hunk 4]
```diff
@@ -67,27 +69,29 @@ mkOrdersExecutor
...
+  -> QSem
   -> BacklogService m
   -> Transactions m era
   -> PoolResolver m
   -> PoolActions
   -> s m ()
-run' logging@Logging{..} backlog@BacklogService{..} txs resolver poolActions =
-  S.repeatM tryAcquire & S.mapM (\case
+run' logging@Logging{..} syncSem backlog@BacklogService{..} txs resolver poolActions =
+  S.before (liftIO $ waitQSem syncSem) $ S.repeatM tryAcquire & S.mapM (\case
```
Comment (oskin1) [inline amm-executor/src/Spectrum/Executor/OrdersExecutor/Process.hs:94]: Why is Sem needed here?

Comment (Bromel777) [reply amm-executor/src/Spectrum/Executor/OrdersExecutor/Process.hs:94]: There is no necessary to start processing orders during sync process

File: amm-executor/src/Spectrum/Executor/EventSink/Handlers/Pools.hs (hunk 3 above)
Comment (Bromel777) [reply amm-executor/src/Spectrum/Executor/EventSink/Handlers/Pools.hs]: fix

File: amm-executor/src/Spectrum/Executor/EventSink/Handlers/Pools.hs (hunk 2 above)
Comment (Bromel777) [reply amm-executor/src/Spectrum/Executor/EventSink/Handlers/Pools.hs]: fix

**Review by oskin1 — APPROVED** [review]

File: amm-executor/src/Spectrum/Executor/EventSink/Handlers/Orders.hs (hunk 1 above)
Comment (Bromel777) [reply amm-executor/src/Spectrum/Executor/EventSink/Handlers/Orders.hs]: fixed

**Pull request merged**